pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/when/impl/BranchConditional.java

```diff
@@ -46,15 +46,40 @@
  * As populated by {@link jenkins.branch.BranchNameContributor}
  */
 public class BranchConditional extends DeclarativeStageConditional<BranchConditional> {
-    private final String compare;
+    private final String pattern;
+    private String comparator;
 
     @DataBoundConstructor
-    public BranchConditional(String compare) {
-        this.compare = compare;
+    public BranchConditional(String pattern) {
+        this.pattern = pattern;
     }
 
+    @Deprecated
     public String getCompare() {
-        return compare;
+        return pattern;
+    }
+
+    public String getPattern() {
+        return pattern;
+    }
+
+    /**
+     * The {@link Comparator} to use.
+     * @return the name of the comparator or null if default.
+     */
+    public String getComparator() {
+        return comparator;
+    }
+
+    @DataBoundSetter
+    public void setComparator(String comparator) {
+        final Comparator c = Comparator.get(comparator, null);
+        //TODO validation
```

Comment:
Yes, please do validation and tests for validation.

Reply:
Likely my bad with C&P from the `TagConditional` class. I was not sure what to implement, so I created a test class for the Comparator:
- https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/351/commits/35c190551ddcf05f4fd956d83940a7c94d62bf8f

Is that what you meant?